src/templates.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2014 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_TEMPLATES_H
+#define BITCOIN_TEMPLATES_H
+
+#include <assert.h>
+#include <cstddef>
+
+template<typename T>
+class Container
```

Comment:
I wanted to allow reviewers to see that we can maintain all the asserts exactly as they are right now, that the functionality with this container is almost identical to the functionality before it.
But I was expecting people to propose other boost/c++11 equivalent classes.
Links to the documentation of your preferred data structure are welcomed.